### PR TITLE
Dynamic cluster roles for Tenant owners

### LINKED ISSUE #524: Dynamic cluster roles for Tenant owners
# Current state

For each Tenant owner in a Tenant specification, Capsule is going to grant `admin` capabilities to the created Namespace resources.

```yaml
apiVersion: capsule.clastix.io/v1beta1
kind: Tenant
metadata:
  name: solar
spec:
  owners:
  - kind: User
    name: alice
  - kind: Group
    name: a-team
status:
  namespaces:
    - solar-development
    - solar-production
```

The resulting RoleBinding will be the following:

```
$: kubectl -n solar-development get rolebindings.rbac.authorization.k8s.io -o wide
NAME                ROLE                                    AGE   USERS   GROUPS   SERVICEACCOUNTS
namespace-deleter   ClusterRole/capsule-namespace-deleter   20s   alice   a-team   
namespace:admin     ClusterRole/admin                       20s   alice   a-team

$: kubectl -n solar-production get rolebindings.rbac.authorization.k8s.io -o wide
NAME                ROLE                                    AGE   USERS   GROUPS   SERVICEACCOUNTS
namespace-deleter   ClusterRole/capsule-namespace-deleter   52s   alice   a-team   
namespace:admin     ClusterRole/admin                       52s   alice   a-team
```

# Desired state

The cluster administrator would like to provide a different role rather than `admin`, and eventually, add different roles to each tenant owner.

A proposed definition could look like the following:

```yaml
apiVersion: capsule.clastix.io/v1beta2
kind: Tenant
metadata:
  name: solar
spec:
  owners:
  - kind: User
    name: alice
    clusterRoleNames:
    - limited-admin
    - namespace-deleter
  - kind: Group
    name: a-team
    clusterRoleNames:
    - limited-admin
```

In that case, users from group `a-team` would be able to manage the Tenant but not deleting any Tenant Namespace: just an example, the feature could provide further multi-tenancy design.

# Implementation

This feature requires a new CRD version, such as `v1beta2` and must be addressed initially using annotations.

```yaml
apiVersion: capsule.clastix.io/v1beta2
kind: Tenant
metadata:
  name: solar
  annotations:
    clusterrolenames.capsule.clastix.io/user.alice: "limited-admin,namespace-deleter"
    clusterrolenames.capsule.clastix.io/group.a-team: "limited-admin,namespace-deleter"
spec:
  owners:
  - kind: User
    name: alice
  - kind: Group
    name: a-team
```

The pattern for the annotation would be the following: `clusterrolenames.capsule.clastix.io/${KIND}.${NAME}=${COMMA_SEPARATED_VALUES}`

- `KIND` would refer to the `/spec/owners/kind` key
- `NAME` would refer to the `/spec/owners/name` key
- `COMMA_SEPARATED_VALUES` would become the new cluster roles that should be bounded to that owner definition

A backward compatibility must be ensured, creating by default the `namespace-deleter` and `namespace:admin` RoleBindings, even with empty annotations.

# Proposal contributors

- @prometherion 
- @MaxFedotov

### PR DESCRIPTION
**Pull request by prometherion**
Closes #524.

~Missing tests, any help on writing them would be superb.~

### CODE REVIEW

**Comment by prometherion** [comment]
Missing documentation, working on that too.

File: controllers/tenant/rolebindings.go [hunk 1]
```diff
@@ -9,16 +9,43 @@ import (
...
+		splitName := strings.Split(owner.Name, ":")
+
+		subject = rbacv1.Subject{
+			Kind:      owner.Kind.String(),
+			Name:      splitName[len(splitName)-1],
+			Namespace: splitName[len(splitName)-2],
+		}
+	} else {
+		subject = rbacv1.Subject{
+			APIGroup: "rbac.authorization.k8s.io",
```
Comment (arriqaaq) [inline controllers/tenant/rolebindings.go:32]: what do you think of making this a constant? I remember before this MR the string "rbac.authorization.k8s.io" was used quite a lot, just avoiding any pitfall in misspell

**Review by arriqaaq — COMMENTED** [review]
lgtm

File: controllers/tenant/rolebindings.go (hunk 1 above)
Comment (prometherion) [reply controllers/tenant/rolebindings.go]: Thanks, addressed with the last amended commits!

**Review by bsctl — APPROVED** [review]
lgtm

**Pull request merged**